### ISSUE TITLE
source from configmap

### ISSUE THREAD

**Issue by kfox1111** [pivotal/kpack#237]
Say I have a very simple function that is small enough to be in a configmap:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: func
data:
  Procfile: |
    web: gunicorn -b 0.0.0.0:8080 app:app
  app.py: |
    from flask import Flask, Response
    import os
    import json

    app = Flask(__name__)

    @app.route('/')
    def get_root():
        js = json.dumps({'hello':'test'})
        resp = Response(js, status=200, mimetype='application/json')
        return resp
    if __name__ == '__main__':
        app.run()
  requirements.txt: |
    flask
    gunicorn
```

It would be great if the Image object could use the configmap as a source for building.

**Comment by matthewmcnew** [comment]
I like the idea. Do you have a specific use case in mind?

**Comment by kfox1111** [comment]
It think it could enable very simple deployment of webhooks or other such glue code.

**Comment by kfox1111** [comment]
Any updates on this?

**Comment by matthewmcnew** [comment]
@kfox1111 No, we haven't invested in this yet. But, would welcome a RFC or PR on it 🙂 .

**Comment by kfox1111** [comment]
I took a look at it and think I might be able to take a stab at it. Honestly, I'm pretty confused on how to add a field to the api though. There's a bunch of generated stuff in there and I can't find any documentation on it.

If someone could point me at some documentation for how to do it, or generate a quick api that adds:

```
configMap:
  name: xxxx
```

to the build.source api then I think I could maybe take it from there.

**Comment by matthewmcnew** [comment]
You should be able to update the types in [pkg/apis/build](https://github.com/pivotal/kpack/tree/master/pkg/apis/build) and regenerate the generated code with `./hack/update-codegen.sh` .

**Comment by kfox1111** [comment]
Ok, I'll have a look. Thanks for the pointer.

**Comment by kfox1111** [comment]
I got a prototype going. Seems to work pretty well. Did take a bit more code then I was expecting. I'm working on getting through the CLA issues. They always take a while to get cleared here.
